src/components/modals/RedemptionInitiatedModal.vue

```diff
@@ -0,0 +1,153 @@
+<script setup lang="ts">
+import { CheckCircleIcon } from '@heroicons/vue/24/outline'
+import BaseButton from '@uicommon/components/BaseButton.vue'
+import BaseModal from '@uicommon/components/BaseModal.vue'
+import { useModals } from '@uicommon/composables/useModals'
+import { formatCurrency } from '@uicommon/utils/numbers'
+import { useElementHover } from '@vueuse/core'
+import dayjs from 'dayjs'
+import { computed, ref, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+import RedemptionInitiatedModalItem from './RedemptionInitiatedModalItem.vue'
+import { EModal, kDefaultDateTimeFormat, kMsInSecond } from '@/constants/ui'
+
+const { t } = useI18n()
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-assignment
+const { visibleModal, hideModal, modalProps } = useModals()
+
+const isPriceVisible = ref(false)
+
+const formattedRedemptionAmount = computed((): number | string => {
+  const rawRedemptionAmount: string = modalProps.value.redemptionAmount
+  return rawRedemptionAmount && Number(rawRedemptionAmount)
+    ? formatCurrency(Number(rawRedemptionAmount))
+    : 0
```

Comment:
Should we ever just return 0?  If `rawRedemptionAmount` doesn't exist, shouldn't an error be (...or already have been) thrown?